pkg/eval/json_evaluator_test.go

```diff
@@ -414,3 +415,87 @@ func TestResolveObjectValue(t *testing.T) {
 		}
 	}
 }
+
+func TestMergeFlags(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name    string
+		current eval.Flags
+		new     eval.Flags
+		want    eval.Flags
+	}{
+		{
+			name:    "both nil",
+			current: eval.Flags{Flags: nil},
+			new:     eval.Flags{Flags: nil},
+			want:    eval.Flags{Flags: map[string]eval.Flag{}},
+		},
+		{
+			name:    "both empty flags",
+			current: eval.Flags{Flags: map[string]eval.Flag{}},
+			new:     eval.Flags{Flags: map[string]eval.Flag{}},
+			want:    eval.Flags{Flags: map[string]eval.Flag{}},
+		},
+
+		{
+			name:    "empty current",
+			current: eval.Flags{Flags: nil},
+			new:     eval.Flags{Flags: map[string]eval.Flag{}},
+			want:    eval.Flags{Flags: map[string]eval.Flag{}},
+		},
+		{
+			name:    "empty new",
+			current: eval.Flags{Flags: map[string]eval.Flag{}},
+			new:     eval.Flags{Flags: nil},
+			want:    eval.Flags{Flags: map[string]eval.Flag{}},
+		},
+		{
+			name: "extra fields on each",
+			current: eval.Flags{Flags: map[string]eval.Flag{
+				"current": {DefaultVariant: "off"},
+			}},
+			new: eval.Flags{Flags: map[string]eval.Flag{
+				"new": {DefaultVariant: "on"},
+			}},
+			want: eval.Flags{Flags: map[string]eval.Flag{
+				"current": {DefaultVariant: "off"},
+				"new":     {DefaultVariant: "on"},
+			}},
+		},
+		{
+			name: "ignore override",
+			current: eval.Flags{Flags: map[string]eval.Flag{
+				"current": {DefaultVariant: "off"},
+			}},
+			new: eval.Flags{Flags: map[string]eval.Flag{
+				"current": {DefaultVariant: "on"},
+				"new":     {DefaultVariant: "on"},
+			}},
+			want: eval.Flags{Flags: map[string]eval.Flag{
+				"current": {DefaultVariant: "off"},
+				"new":     {DefaultVariant: "on"},
+			}},
+		},
```

Comment:
Am I misunderstanding, or is this the opposite of what we decided?

Based on the names, it seems like the newest flag does not win, the oldest does.

Reply:
I agree @James-Milligan, will rename it.

Reply:
Done.

Reply:
@moshebe I'm sorry to trouble you more, this is a great PR, the code and tests look excellent. I think we probably should do "last wins", instead of "first wins", like we seemed to agree upon in the [issue](https://github.com/open-feature/flagd/issues/25#issuecomment-1181941174). I think this will be especially important in the K8s CRD use-case I give [here](https://github.com/open-feature/flagd/issues/25#issuecomment-1181965242).

I think listing flags with increasing specificity will be the intuitive pattern, and in that case having the last win also seems intuitive:

```
annotations:
        openfeature.dev: "enabled"
        openfeature.dev/featureflagconfiguration: ["common-flags", "specific-flags"]   # we want "specific-flags" to win here.
```

Please let me know if you think first wins is better for a specific reason!

Reply:
@toddbaert nothing specific with this methods, just wanted to set a deterministic behavior.
Thanks for your feedback, glad to contribute!

Made the changes you suggested and updated the README and flag help description.
Please let me know what do you think.